Sources/SwiftInspectorCore/Tests/PropertyAnalzyerSpec.swift

```diff
@@ -0,0 +1,468 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import Foundation
+
+@testable import SwiftInspectorCore
+
+final class PropertyAnalyzerSpec: QuickSpec {
+  override func spec() {
+    var fileURL: URL!
+    var sut = PropertyAnalyzer(typeName: "FakeType")
+
+    beforeEach {
+      sut = PropertyAnalyzer(typeName: "FakeType")
+    }
+
+    afterEach {
+      guard let fileURL = fileURL else {
+        return
+      }
+      try? Temporary.removeItem(at: fileURL)
+    }
+
+    describe("TypeWithPropInfo.merge(other:)") {
+      let type1 = TypeWithPropInfo(
+        name: "MyType",
+        properties: [
+          TypeWithPropInfo.PropInfo(name: "thing", typeAnnotation: "String", comment: "", access: .public, scope: .instance)
+      ])
+      let type2 = TypeWithPropInfo(
+        name: "MyType",
+        properties: [
+          TypeWithPropInfo.PropInfo(name: "foo", typeAnnotation: "Int", comment: "", access: .public, scope: .instance)
+      ])
+      let type3 = TypeWithPropInfo(
+        name: "AnotherType",
+        properties: [
+          TypeWithPropInfo.PropInfo(name: "foo", typeAnnotation: "Int", comment: "", access: .public, scope: .instance)
+      ])
+
+      context("when both types have the same name") {
+        let result = try? type1.merge(with: type2)
+        it("succeeds") {
+          expect(result?.name) == "MyType"
+        }
+
+        it("has merged props") {
+          let set = Set(result?.properties ?? [])
+          let expectedSet: Set<TypeWithPropInfo.PropInfo> = [
+            TypeWithPropInfo.PropInfo(name: "thing", typeAnnotation: "String", comment: "", access: .public, scope: .instance),
+            TypeWithPropInfo.PropInfo(name: "foo", typeAnnotation: "Int", comment: "", access: .public, scope: .instance)
+          ]
+          expect(set) == expectedSet
+        }
+      }
+
+      context("when the types don't match") {
+        it("fails to merge and asserts") {
+          expect { try type1.merge(with: type3) }.to(throwError())
+        }
+      }
+
+      context("when the other type is nil") {
+        it("returns the original") {
+          expect(try? type1.merge(with: nil)) == type1
+        }
+      }
+    }
+
+    describe("analyze(fileURL:)") {
+      context("when there are no properties") {
+        beforeEach {
+          let content = """
+                        public final class FakeType {}
+                        """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("returns type info with empty property list") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.properties).to(beEmpty())
+        }
+      }
+
+      context("when there is a property") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
```

Comment:
What would happen if I had this (valid) Swift code?

```swift
class FakeType {

  let hello: Int = 1

  class FakeType {
    let goodbye: String = ""
  }
}
```

Reply:
Here's how I handled that case for type location but it's a bit different here https://github.com/fdiaz/SwiftInspector/blob/a173d02d773af433cc9e36c5797f58a5f7735445/Sources/SwiftInspectorCore/Tests/TypeLocationAnalyzerSpec.swift#L295-L317

Reply:
@bachand I would argue that is very poor API design 😅 

I don't know, what do you think we should do here? I can handle the case just fine that you posted below where the nested class type is different from the outer class, but I don't know how to handle the nested class having the same name as the outer class. That just feels *super* rare to the point we don't need to cover it.

Reply:
@bachand yeah this is a great point... I don't currently handle this situation well - we will just merge all of the properties into one list. Do you have suggestions of how we should handle this? I could append the parent type name to the child type like:
```swift
FakeType.FakeType
```
to disambiguate. You would then need to pass in `FakeType.FakeType` in order to get property information from a nested type.

Reply:
The example I gave is pretty contrived.

One that is maybe (a bit) more realistic could look like this.

```swift
class MyClass {

  let hello: Int = 1

  class NestedType {
    let goodbye: String = ""
  }
}

class YourClass {

  let foo: Int = 1

  class NestedType {
    let bar: String = ""
  }
}
```

Ideally we could look up information on `MyClass.NestedType` and `YourClass.NestedType` independently.

Appending the parent type name definitely seems like the Right™ way to do this. I didn't think of that!

Reply:
I just tried doing this and I'm not even sure how to accomplish what I suggested. Very open to suggestions on how to handle this!

Reply:
Haha, yeah this sounds hard to me. I think you'd have to keep track of the parent type(s) as you traverse the nodes.

I can live with this limitation for now. It could be nice to have a test to show how we handle this case. Then we could add a comment to that test linking to an issue where we describe the ideal behavior.

cc: @fdiaz

Reply:
I think this is fine for now. Let's keep it as is and if we see problems in the real world or a use case where we need to handle this we can build on top of this.